include/platform/mir/graphics/platform.h

```diff
@@ -164,6 +164,10 @@ enum PlatformPriority : uint32_t
     supported = 128,    /**< Capable of providing a functioning Platform on this device,
                          *    possibly with degraded performance or features.
                          */
+    hosted = 192,       /**< Capable of providing a fully-featured Platform on this device,
+                         *   running nested under some other display server rather than with
+                         *   exclusive hardware access.
+                         */
```

Comment:
Should the Wayland platform also return `hosted`? Currently we have:
```c++
    return mpw::connection_options_supplied(options) ? mg::PlatformPriority::best :
           mg::PlatformPriority::unsupported;
```

Reply:
I think in this case Wayland should return `best` - this isn't an auto-probe, this is “the user has explicitly set the command-line option to use the Wayland platform”.

Maybe what should happen here is the Wayland platform should return “the user has set a Wayland platform option, so Wayland is the only sensible option” value? Or do we want to be able to handle “*if* the Wayland platform is autoselected, have it use this option” cases?

Maybe something for a followup discussion?